python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -74,6 +73,7 @@ class Node(object):
     thread-safe usage.
     """
     __metaclass__ = abc.ABCMeta
+    __slots__ = ('_index_cache_dict', '_parent', '_read_lock', '_write_lock', '_dirty', '_registry', '_data_dict')
```

Comment:
While you can indeed add `__dict__` to `__slots__` it rather undoes most (if not all) of the point of using `__slots__` in the first place and might in fact make it even worse. `__slots__` is something which should be fallen back on after careful memory profiling. What memory reduction did you see after this change?